store.go

```diff
@@ -2402,11 +2470,28 @@ func (s *store) ContainerDirectory(id string) (string, error) {
 		return "", err
 	}
 
+	rootUID, rootGID, err := idtools.GetRootUIDGID(s.uidMap, s.gidMap)
+	if err != nil {
+		return "", err
+	}
+
 	middleDir := s.graphDriverName + "-containers"
 	gcpath := filepath.Join(s.GraphRoot(), middleDir, id, "userdata")
-	if err := os.MkdirAll(gcpath, 0700); err != nil {
+	if err := idtools.MkdirAllAs(gcpath, 0700, rootUID, rootGID); err != nil {
 		return "", err
 	}
+
+	if rootUID != 0 {
+		userdata := filepath.Join(s.RunRoot(), id, "userdata")
```

Comment:
This seems to break the implicit promise that the contents of the `ContainerDirectory()` will survive a reboot.

Reply:
fixed